autogpt_platform/backend/backend/blocks/llm.py

```diff
@@ -1451,7 +1451,19 @@ async def _summarize_chunk(
             credentials=credentials,
         )
 
-        return llm_response["summary"]
+        summary = llm_response["summary"]
+
+        # Validate that the LLM returned a string and not a list or other type
+        if not isinstance(summary, str):
+            raise ValueError(
+                f"LLM generation failed: Expected a string summary, but received {type(summary).__name__}. "
+                f"The language model incorrectly formatted its response. "
+                f"This may be due to the model interpreting the '{input_data.style}' style "
+                f"as a request for a list format instead of a single text summary. "
+                f"Received value: {summary}"
```

Comment:
we can use truncate.py + json.py to dump and make sure the content of the summary is size-limited.